File: 프로그래머스/0/120892. 암호 해독/암호 해독.java
class Solution {
    public String solution(String cipher, int code) {
        String answer = "";
        int num = cipher.length() % code == 0 ? cipher.length() / code :
        (cipher.length()/code) - 1;
        
        for(int i=code-1;i<cipher.length();i+=code){
            answer += cipher.charAt(i);          
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120892. 암호 해독/암호 해독.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

char* solution(const char* cipher, int code) {
    char* answer = (char*)malloc(strlen(cipher)+1/code);

    int len = 0;
    
    for(int i=code;i<=strlen(cipher);i+=code){
        answer[len++] = cipher[i-1];
        }
    
    answer[len] = NULL;
    return answer;
}